ICE: in visit, at rust/typecheck/rust-hir-type-check-enumitem.cc:204

<!--
File: /tmp/im/20D26B9876141A8F6E0ACF940AD61521BB68AF5B1F2AF3630F6DB35DD986C745.rs
-->

auto-reduced (treereduce-rust):
````rust
//@compile-flags: -frust-incomplete-and-experimental-compiler-do-not-use
enum E {
    A = {
        enum Foo {
            Bar(isize),
            Baz,
        }
        0
    },
}

````

original:
````rust
// check-pass

enum E {
    A = {
        enum Foo {
    Bar(isize),
    Baz,
}
        0
    }
}

fn main() {}

````

Version information:

765121736dfe3f5b319bbe9837880deda326394e

Possibly related line of code:
https://github.com/Rust-GCC/gccrs/blob/765121736dfe3f5b319bbe9837880deda326394e/gcc/rust/typecheck/rust-hir-type-check-enumitem.cc#L198-L210

Command:
`/home/matthias/vcs/github/gccrs/gccrs-build/gcc/crab1 -frust-incomplete-and-experimental-compiler-do-not-use`

<details><summary><strong>Program output</strong></summary>
<p>

```
/tmp/icemaker_global_tempdir.NKudfJBTYGQf/rustc_testrunner_tmpdir_reporting.5SHLYwQhKAeY/mvce.rs:4:17: error: unknown reference for resolved name: ‘isize’
    4 |             Bar(isize),
      |                 ^~~~~
crab1: internal compiler error: in visit, at rust/typecheck/rust-hir-type-check-enumitem.cc:204
0x891647 Rust::Resolver::TypeCheckEnumItem::visit(Rust::HIR::EnumItemTuple&)
	../../gcc/rust/typecheck/rust-hir-type-check-enumitem.cc:204
0xfc0158 Rust::Resolver::TypeCheckEnumItem::Resolve(Rust::HIR::EnumItem&, long)
	../../gcc/rust/typecheck/rust-hir-type-check-enumitem.cc:43
0xf83021 Rust::Resolver::TypeCheckItem::visit(Rust::HIR::Enum&)
	../../gcc/rust/typecheck/rust-hir-type-check-item.cc:352
0xf7dbb7 Rust::Resolver::TypeCheckItem::Resolve(Rust::HIR::Item&)
	../../gcc/rust/typecheck/rust-hir-type-check-item.cc:59
0xfb995b ???
	../../gcc/rust/typecheck/rust-hir-type-check-stmt.h:39
0xfb9f15 Rust::Resolver::TypeCheckStmt::Resolve(Rust::HIR::Stmt&)
	../../gcc/rust/typecheck/rust-hir-type-check-stmt.cc:34
0xfaba54 Rust::Resolver::TypeCheckExpr::visit(Rust::HIR::BlockExpr&)
	../../gcc/rust/typecheck/rust-hir-type-check-expr.cc:594
0xfaa2b8 Rust::Resolver::TypeCheckExpr::Resolve(Rust::HIR::Expr&)
	../../gcc/rust/typecheck/rust-hir-type-check-expr.cc:47
0xfbc509 Rust::Resolver::TypeCheckEnumItem::visit(Rust::HIR::EnumItemDiscriminant&)
	../../gcc/rust/typecheck/rust-hir-type-check-enumitem.cc:114
0xfc018f Rust::Resolver::TypeCheckEnumItem::Resolve(Rust::HIR::EnumItem&, long)
	../../gcc/rust/typecheck/rust-hir-type-check-enumitem.cc:51
0xf83021 Rust::Resolver::TypeCheckItem::visit(Rust::HIR::Enum&)
	../../gcc/rust/typecheck/rust-hir-type-check-item.cc:352
0xf7dbb7 Rust::Resolver::TypeCheckItem::Resolve(Rust::HIR::Item&)
	../../gcc/rust/typecheck/rust-hir-type-check-item.cc:59
0xf1518b Rust::Resolver::TypeResolution::Resolve(Rust::HIR::Crate&)
	../../gcc/rust/typecheck/rust-hir-type-check.cc:75
0xce8f73 Rust::Session::compile_crate(char const*)
	../../gcc/rust/rust-session-manager.cc:685
Please submit a full bug report, with preprocessed source (by using -freport-bug).
Please include the complete backtrace with any bug report.
See <https://gcc.gnu.org/bugs/> for instructions.

```

</p>
</details>

--- a/gcc/testsuite/rust/compile/issue-3555.rs
+++ b/gcc/testsuite/rust/compile/issue-3555.rs
@@ -1,0 +1,14 @@
+#![feature(no_core)]
+#![no_core]
+
+fn main() {
+    match SomeStruct(2) {
+        StructConst1(_) => {}
+
+        _ => {}
+    }
+
+    struct SomeStruct(u8);
+
+    const StructConst1: SomeStruct = SomeStruct(1);
+}

--- a/gcc/testsuite/rust/compile/issue-3591.rs
+++ b/gcc/testsuite/rust/compile/issue-3591.rs
@@ -1,0 +1,8 @@
+#![feature(no_core)]
+#![no_core]
+
+fn main() {
+    match (0, 92, 29) {
+        (pat, ..) => {}
+    }
+}

--- a/gcc/testsuite/rust/compile/issue-3597.rs
+++ b/gcc/testsuite/rust/compile/issue-3597.rs
@@ -1,0 +1,11 @@
+#![feature(no_core)]
+#![no_core]
+
+enum E {
+    A = {
+        enum F {  }
+        0
+    }
+}
+
+fn main() {}

--- a/gcc/testsuite/rust/compile/issue-3651.rs
+++ b/gcc/testsuite/rust/compile/issue-3651.rs
@@ -1,0 +1,12 @@
+#![feature(no_core)]
+#![no_core]
+enum E {
+    A = {
+        enum F {
+            Ai32 = -1,
+        }
+        0
+    },
+}
+
+fn main() {}

--- a/gcc/testsuite/rust/compile/issue-3670.rs
+++ b/gcc/testsuite/rust/compile/issue-3670.rs
@@ -1,0 +1,11 @@
+#![feature(no_core)]
+#![no_core]
+enum E {
+    A = {
+        enum Foo {
+            Bar(isize),
+            Baz,
+        }
+        0
+    },
+}

--- a/gcc/testsuite/rust/compile/issue-3672.rs
+++ b/gcc/testsuite/rust/compile/issue-3672.rs
@@ -1,0 +1,12 @@
+// { dg-options "-w" }
+#![feature(no_core)]
+#![no_core]
+struct Y(usize, usize);
+
+fn distinct_variant() {
+    let y = Y(1, 2);
+
+    let a = match y {
+        Y(a, _) => a,
+    };
+}

